### DISCUSSION TITLE
What should I do if I want to use vue-router through cdn in my website?

### DISCUSSION THREAD

**Discussion by yonecdeng** [vuejs/router#2241] [Help and Questions] [answered]
### Reproduction

1

### Steps to reproduce the bug

1. vite project
2. external 'vue-router' when you build your project.
3. inject `https://cdnjs.cloudflare.com/ajax/libs/vue-router/4.3.2/vue-router.esm-browser.min.js` in your html.

### Expected behavior

run normally

### Actual behavior

throw error in chrome 
<img width="979" alt="image" src="https://github.com/vuejs/router/assets/73653404/213ac7a5-00a2-4d35-b52e-132b1131304a">

**Comment by posva** [comment, marked as answer]
You will need https://github.com/vuejs/router/pull/2242

FYI I wouldn't recommend using CDN versions for apps because you import **everything** and lose the benefits of tree shaking

**Reply by yonecdeng** [reply]
thanks your answer. but i have to extract public dependencies for multiple projects.
